File: MusicManagerAPI/MusicManagerAPI-client/src/main/java/music/manager/MusicTypeClient.java
package music.manager;

import music.manager.model.UserMusicStatusDTO;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

/**
 * Created by ldalzotto on 25/04/2017.
 */
@FeignClient("music-manager-api")
public interface MusicTypeClient {

    @RequestMapping(method = RequestMethod.POST, path = "/user")
    public ResponseEntity<?> postUser(@RequestBody UserMusicStatusDTO userMusicStatusDTO);

    @RequestMapping(method = RequestMethod.GET, path = "/user/username/{username}")
    public ResponseEntity<UserMusicStatusDTO> getUserFromUsername(@PathVariable("username") String username);

    @RequestMapping(method = RequestMethod.DELETE, path = "/user/username/{username}")
    public ResponseEntity<?> deleteUser(@PathVariable("username") String username);

    @RequestMapping(method = RequestMethod.GET, path = "/music/musicplatform/{musicplatform}/ressource-url")
    public ResponseEntity<String> getRessourceUrlFromUrlAndMusicplatform(@RequestParam("url") String url, @PathVariable("musicplatform") String musicplatform);

    @RequestMapping(method = RequestMethod.POST, path = "/music/username/{username}/from-source-url")
    public ResponseEntity<Boolean> postMusicFromSiteurl(@PathVariable("username") String username, @RequestParam("sourceurl") String sourceUrl);

}

File: MusicManagerAPI-client/src/scala/com/ldz/music/manager/MusicTypeClient.scala
package com.ldz.music.manager

import com.ldz.music.manager.model.{MusicTypeDTO, UserMusicStatusDTO}
import com.ldz.enumeration.ExternalMusicKey
import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation._

/**
  * Created by Loic on 20/06/2017.
  */
@FeignClient("music-manager-api")
trait MusicTypeClient {
  @RequestMapping(method = Array(RequestMethod.POST), path = Array("/music/user"))
  def postUser(@RequestBody userMusicStatusDTO: UserMusicStatusDTO): ResponseEntity[_]

  @RequestMapping(method = Array(RequestMethod.GET), path = Array("/music/user/username/{username}"))
  def getUserFromUsername(@PathVariable("username") username: String): ResponseEntity[UserMusicStatusDTO]

  @RequestMapping(method = Array(RequestMethod.DELETE), path = Array("/music/user/username/{username}"))
  def deleteUser(@PathVariable("username") username: String): ResponseEntity[_]

  @RequestMapping(method = Array(RequestMethod.GET), path = Array("/music/musicplatform/{musicplatform}/ressource-url"))
  def getMusicParametersFromUrlAndMusicplatform(@RequestParam("url") url: String, @PathVariable("musicplatform") musicplatform: String): ResponseEntity[MusicTypeDTO]

  @RequestMapping(method = Array(RequestMethod.PUT), path = Array("/music/user/username/{username}"))
  def postMusicFromSiteurl(@PathVariable("username") username: String, @RequestBody musicType: MusicTypeDTO): ResponseEntity[MusicTypeDTO]
}
